### PR TITLE
[Feature/agent_framework] Registers root agent with an agentId in ToolSteps

### PR DESCRIPTION
**Pull request by owaiskazi19**
### Description
This PR:
1. Registers an agent with an `agent_id`. Ref: https://github.com/opensearch-project/flow-framework/issues/194#issuecomment-1837266521
2. Integrates the new Util method for ToolSteps and RegisterAgentSteps.

~TODO:~
~https://github.com/opensearch-project/flow-framework/issues/241~

### Issues Resolved
Closes https://github.com/opensearch-project/flow-framework/issues/194

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/flowframework/workflow/RegisterAgentStep.java [hunk 1]
```diff
@@ -240,7 +231,7 @@ private String getLlmModelId(Map<String, String> previousNodeInputs, Map<String,
 
     private LLMSpec getLLMSpec(String llmModelId, Map<String, String> llmParameters) {
         if (llmModelId == null) {
-            throw new IllegalArgumentException("model id for llm is null");
+            throw new FlowFrameworkException("model id for llm is null", RestStatus.BAD_REQUEST);
```
Comment (dbwiddis) [inline src/main/java/org/opensearch/flowframework/workflow/RegisterAgentStep.java:234]: Can we pass workflow ID and step ID to this method to include in the exception? (Can handle in a future PR)

File: src/main/java/org/opensearch/flowframework/workflow/RegisterAgentStep.java [hunk 2]
```diff
@@ -101,84 +101,55 @@ public void onFailure(Exception e) {
...
-        if (llmModelId == null) {
-            registerAgentModelFuture.completeExceptionally(
-                new FlowFrameworkException("llm model id is not provided", RestStatus.BAD_REQUEST)
-            );
-            return registerAgentModelFuture;
-        }
+            // Case when modelId is not present at all
+            if (llmModelId == null) {
+                registerAgentModelFuture.completeExceptionally(
+                    new FlowFrameworkException("llm model id is not provided", RestStatus.BAD_REQUEST)
```
Comment (dbwiddis) [inline src/main/java/org/opensearch/flowframework/workflow/RegisterAgentStep.java:146]: Please add workflowID and step (node) ID to this error message. (Can handle in a future PR)

**Review by dbwiddis — APPROVED** [review]
LGTM. Would like error messages to be more verbose but that can be handled in another PR

File: src/main/java/org/opensearch/flowframework/workflow/RegisterAgentStep.java (hunk 1 above)
Comment (dbwiddis) [reply src/main/java/org/opensearch/flowframework/workflow/RegisterAgentStep.java:226]: https://github.com/opensearch-project/flow-framework/issues/244

**Review by joshpalis — APPROVED** [review]

**Pull request merged**